AlphaWallet/Transfer/Controllers/TransactionConfigurator.swift

```diff
@@ -85,10 +85,18 @@ class TransactionConfigurator {
                 return token.contractAddress
             }
         }()
+        //TODO transaction.value should only ever be the attached native currency, not the erc20 amount as that is included in the data
```

Comment:
Not a TODO, but a comment for the "why" (which is good) right?

Reply:
Do you want to extract it out into a property in `TransferType`?

So you can write here:

```
        let value: BigInt = transaction.nativeCryptoValue
```

But isn't `transaction.value` the native crypto value even if it's an ERC 20 transfer? The token amount is encoded into `data`? Or is it because the gas estimator only looks at the ETH value even if the data is not 0x0? (e.g. smart contract call — eg. ERC20 transfer — which also pays ETH) and we are trying to workaround the estimator?

Reply:
No, it's a TODO because it is incorrectly placing the erc20 amount in there.

Reply:
Ah. I see. How did it ever work! Aargh.